openwisp_users/admin.py

```diff
@@ -122,16 +124,51 @@ def has_add_permission(self, request, obj=None):
 class EmailRequiredMixin(forms.ModelForm):
     email = forms.EmailField(label=_('Email'), max_length=254, required=True)
 
+    def validate_user_groups(self, data):
+        is_staff = data.get('is_staff')
+        groups = data.get('groups')
+        if is_staff and not groups:
+            raise ValidationError(
+                {'groups': _('A staff user must belong to a group, please select one.')}
+            )
+
 
 class UserCreationForm(EmailRequiredMixin, BaseUserCreationForm):
+    phone_number = PhoneNumberField(widget=forms.TextInput(), required=False)
+
     class Meta(BaseUserCreationForm.Meta):
-        fields = ['username', 'email', 'password1', 'password2', 'is_staff']
-        fields_superuser = fields[:] + ['is_superuser']
-        fieldsets = ((None, {'classes': ('wide',), 'fields': fields}),)
+        fields = [
+            'username',
+            'email',
+            'password1',
+            'password2',
+        ]
+        personal_fields = ['first_name', 'last_name', 'phone_number']
+        fieldsets = (
+            (None, {'classes': ('wide',), 'fields': fields}),
+            ('Personal Info', {'classes': ('wide',), 'fields': personal_fields}),
+            (
+                'Permissions',
+                {'classes': ('wide',), 'fields': ['is_active', 'is_staff', 'groups']},
+            ),
+        )
         fieldsets_superuser = (
-            (None, {'classes': ('wide',), 'fields': fields_superuser}),
+            (None, {'classes': ('wide',), 'fields': fields}),
+            ('Personal Info', {'classes': ('wide',), 'fields': personal_fields}),
+            (
+                'Permissions',
+                {
+                    'classes': ('wide',),
+                    'fields': ['is_active', 'is_staff', 'is_superuser', 'groups'],
+                },
+            ),
         )
 
+    def clean(self):
+        cleaned_data = super().clean()
+        self.validate_user_groups(cleaned_data)
+        return cleaned_data
```

Comment:
there's repetition of this in both forms, you should not do that.
See again my previous review, I left a suggestion that will solve this:
https://github.com/openwisp/openwisp-users/pull/133#pullrequestreview-442572930